test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceMappingProviderTests.cs

```diff
@@ -410,5 +410,373 @@ public void AddOrUpdatePackageSourceMapping_WithAClearItem_WithANewPattern_AddIn
+
+            tempMappings.Add(testMappingItemAdd);
+            tempMappings.Remove(testMappingItem2);
+
+            sourceMappingProvider.SavePackageSourceMappings(tempMappings);
+
+            var result = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSourceMapping>
+        <clear />
+        <packageSource key=""nuget.org"">
+            <package pattern=""0"" />
+            <package pattern=""1"" />
+            <package pattern=""2"" />
+        </packageSource>
+        <packageSource key=""newSource"">
+            <package pattern=""stuff"" />
+        </packageSource>
+    </packageSourceMapping>
+</configuration>";
+
+            result.Replace("\r\n", "\n")
+                .Should().BeEquivalentTo(
+                File.ReadAllText(configPath1).Replace("\r\n", "\n"));
+        }
+
+        [Fact]
+        public void SavePackageSourceMappings_PackageSourceMappingDisabled_AddMapping()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+</configuration>");
+
+            List<PackagePatternItem> testPackagePatternItems = new List<PackagePatternItem>();
+
+            List<PackagePatternItem> packagePatternItems = new List<PackagePatternItem>();
+            for (int i = 0; i < 3; i++)
+            {
+                PackagePatternItem tempPackagePatternItem = new PackagePatternItem(i.ToString());
+                packagePatternItems.Add(tempPackagePatternItem);
+            }
+            PackageSourceMappingSourceItem tempMapping = new PackageSourceMappingSourceItem("tempSource", packagePatternItems);
+            List<PackageSourceMappingSourceItem> tempMappings = new List<PackageSourceMappingSourceItem>();
+            tempMappings.Add(tempMapping);
+
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var sourceMappingProvider = new PackageSourceMappingProvider(settings);
+            sourceMappingProvider.SavePackageSourceMappings(tempMappings);
+
+            var result = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSourceMapping>
+    <packageSource key=""tempSource"">
+        <package pattern=""0"" />
+        <package pattern=""1"" />
+        <package pattern=""2"" />
+      </packageSource>
+  </packageSourceMapping>
+</configuration>";
+
+            result.Replace("\r\n", "\n")
+                .Should().BeEquivalentTo(
+                File.ReadAllText(configPath1).Replace("\r\n", "\n"));
+        }
+
+        [Fact]
+        public void SavePackageSourceMappings_NewPatternExistingSource_Add()
```

Comment:
Once you've addressed the concern about the tests with the same implementation, I think you'll need to dedup whatever the outcome there is to this, because I imagine one of those tests will be equivalent ot this one.

Reply:
This one is a bit different from SavePackageSourceMappings_WithNewMappings_Add() because it adds one new pattern to a source with existing mappings, but SavePackageSourceMappings_WithNewMappings_Add() adds a new mapping to a new source

Reply:
Resolving this.

Both my comments seem to be referring to the same misreading of a test case.